bundles/de.uka.ipd.sdq.simucomframework/src/de/uka/ipd/sdq/simucomframework/model/SimuComModel.java

```diff
@@ -70,14 +67,20 @@ public SimuComModel(final SimuComConfig config, final SimuComStatus status, fina
             final boolean isRemoteRun, IResourceTableManager resourceTableManager) {
         this(config, status, factory, isRemoteRun, null, resourceTableManager);
     }
-
+    
     public SimuComModel(final SimuComConfig config, final SimuComStatus status, final ISimEngineFactory factory,
             final boolean isRemoteRun, final ProbeFrameworkContext probeFrameworkContext, IResourceTableManager resourceTableManager) {
+        this(config, status, factory, isRemoteRun, probeFrameworkContext, resourceTableManager, null);
+    }
+
+    public SimuComModel(final SimuComConfig config, final SimuComStatus status, final ISimEngineFactory factory,
+            final boolean isRemoteRun, final ProbeFrameworkContext probeFrameworkContext, IResourceTableManager resourceTableManager,
+            ResourceRegistry resourceRegistry) {
         this.config = config;
         this.simulationEngineFactory = factory;
         factory.setModel(this);
         this.simControl = factory.createSimulationControl();
-        resourceRegistry = new ResourceRegistry(this);
+        this.resourceRegistry = resourceRegistry == null ? new ResourceRegistry(this) : resourceRegistry;

```

Comment:
why 'error' handling here? if someone passes a null reference this is an indication that something went wrong at another spot; if you create a new ResourceRegistry here in case of an error, then the originally problem is covered;
I vote for anly assigning the passed resourceRegistry instead of the null check + new creation

Reply:
In principal I fully agree with you. Unfortunately it is legacy behavior of SimuCom to have the ResourceRegistry initialized in the constructor of the model. In SimuLizar we now create the registry outside of the model. Once we get rid of SimuCom support we could remove this fallback. Otherwise we would need to adapt SimuCom to the new interface, an effort I would like to omit.

Reply:
I added some explaining comments